tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapEntryListenerTest.java

```diff
@@ -59,176 +73,200 @@
     public int maxEntryListenerDelayMs = 0;
     public int minEntryListenerDelayMs = 0;
 
-    //check these add up to 1
-    public double writeProb = 0.4;
+    public double putProb = 0.4;
     public double evictProb = 0.2;
     public double removeProb = 0.2;
     public double deleteProb = 0.2;
 
-    //check these add up to 1   (writeProb is split up into sub styles)
-    public double writeUsingPutProb = 0.5;
-    public double writeUsingPutIfAbsent = 0.25;
-    public double replaceProb = 0.25;
+    public double putUsingPutIfAbsentProb = 0.25;
+    public double putUsingReplaceProb = 0.25;
+
+    private final ScrambledZipfianGenerator keysZipfian = new ScrambledZipfianGenerator(keyCount);
+    private final OperationSelectorBuilder<Operation> operationSelectorBuilder = new OperationSelectorBuilder<Operation>();
+    private final OperationSelectorBuilder<PutOperation> putOperationSelectorBuilder
+            = new OperationSelectorBuilder<PutOperation>();
 
     private String[] values;
-    private TestContext testContext;
-    private HazelcastInstance targetInstance;
-    private EntryListenerImpl listener;
-    private ScrambledZipfianGenerator kesyZipfian = new ScrambledZipfianGenerator(keyCount);
-    private IMap<Object, Object> map;
+    private EntryListenerImpl<Integer, String> listener;
+    private IList<EventCount> eventCounts;
+    private IList<EntryListenerImpl<Integer, String>> listeners;
+    private IMap<Integer, String> map;
 
     @Setup
     public void setup(TestContext testContext) throws Exception {
-        this.testContext = testContext;
-        targetInstance = testContext.getTargetInstance();
+        HazelcastInstance targetInstance = testContext.getTargetInstance();
+
         values = StringUtils.generateStrings(valueCount, valueLength);
+        listener = new EntryListenerImpl<Integer, String>(minEntryListenerDelayMs, maxEntryListenerDelayMs);
+
+        eventCounts = targetInstance.getList(basename + "eventCount");
+        listeners = targetInstance.getList(basename + "listeners");
 
         map = targetInstance.getMap(basename);
-        listener = new EntryListenerImpl(minEntryListenerDelayMs, maxEntryListenerDelayMs);
         map.addEntryListener(listener, true);
+
+        operationSelectorBuilder
```

Comment:
I find this part a bit confusing because one is operation (so very generic) and the other one is putoperations which is very specific. I think they are both specific. Can you think about the name 'operationSelectorBuilder'?